lib/pages/labels/label_bloc.dart

```diff
@@ -47,13 +47,23 @@ class LabelBloc implements BlocBase {
     _loadLabels();
   }
 
-  void checkIfLabelExist(Label label) async {
-    _labelDB.isLabelExits(label).then((isExist) {
-      _labelExistController.sink.add(isExist);
+  void createOrExists(Label label) async {
+    _labelDB.isLabelExists(label).then((exist) {
+      _labelExistController.sink.add(exist);
+      if (!exist) {
+        _labelDB.updateLabels(label);
+        _loadLabels();
+      }
     });
   }
 
   void updateColorSelection(ColorPalette colorPalette) {
     _colorController.sink.add(colorPalette);
   }
+
+  void deleteLabel(int labelID) async {
```

Comment:
Can you please write unit tests for this?

Reply:
test case `Delete label in label list test` covers this.

Reply:
I cannot find the tests

Reply:
the test is located in `WhatTodo\test\label_bloc_test.dart`

Reply:
did you find it ?